subgraphs/morpho/src/mapping/morpho-compound/morphoCompound.ts

```diff
@@ -297,3 +315,98 @@ export function handleReserveFactorSet(event: ReserveFactorSet): void {
 // eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/no-empty-function
 export function handleReserveFeeClaimed(event: ReserveFeeClaimed): void {}
+
+// Update $COMP distributed rewards
+export function updateCompoundRewards(market: Market): void {
+  // DEPOSIT first because it alphabetizes
+  const rewardTokens = [
+    getOrCreateRewardToken(COMP_ADDRESS, RewardTokenType.DEPOSIT).id,
+    getOrCreateRewardToken(COMP_ADDRESS, RewardTokenType.VARIABLE_BORROW).id,
+  ];
+  const amounts = [BigInt.zero(), BigInt.zero()];
+  const amountsUSD = [BigDecimal.zero(), BigDecimal.zero()];
+
+  // get token amount in Compound market
+  const cTokenContract = CToken.bind(Address.fromBytes(market.id));
+  const tryCTokenAmount = cTokenContract.try_totalSupply();
+  const tryBalanceOf = cTokenContract.try_balanceOf(MORPHO_COMPOUND_ADDRESS);
+  if (tryCTokenAmount.reverted || tryBalanceOf.reverted) {
+    log.error(
+      "[updateCompoundRewards] cTokenContract.try_totalSupply() or cTokenContract.try_balanceOf(MORPHO_COMPOUND_ADDRESS) reverted for market: {}",
+      [market.id.toHexString()]
+    );
+    return;
+  }
+  const morphoShare = tryBalanceOf.value
+    .toBigDecimal()
+    .div(tryCTokenAmount.value.toBigDecimal());
+
+  // get COMP reward rate per side
+  const comptrollerContract = Comptroller.bind(COMPTROLLER_ADDRESS);
+  const trySupplySpeed = comptrollerContract.try_compSupplySpeeds(
+    Address.fromBytes(market.id)
+  );
+  const tryBorrowSpeed = comptrollerContract.try_compBorrowSpeeds(
+    Address.fromBytes(market.id)
+  );
+
+  if (trySupplySpeed.reverted || tryBorrowSpeed.reverted) {
+    log.error(
+      "[updateCompoundRewards] comptrollerContract.try_compSupplySpeeds(market.id) or comptrollerContract.try_compBorrowSpeeds(market.id) reverted for market: {}",
+      [market.id.toHexString()]
+    );
+    return;
+  }
+
+  // get COMP price in USD
+  const protocol = LendingProtocol.load(market.protocol);
+  if (!protocol) {
+    log.error("[updateCompoundRewards] protocol not found for protocol: {}", [
+      market.protocol.toHexString(),
+    ]);
+    return;
+  }
+  let priceUSD = BigDecimal.zero();
+  if (protocol._oracle) {
+    const oracleContract = CompoundOracle.bind(
+      Address.fromBytes(protocol._oracle!)
+    );
+    const tryPriceUSD = oracleContract.try_getUnderlyingPrice(CCOMP_ADDRESS);
+    const bdFactor = exponentToBigDecimal(DEFAULT_DECIMALS);
+    if (!tryPriceUSD.reverted) {
+      priceUSD = tryPriceUSD.value.toBigDecimal().div(bdFactor);
+    }
+  }
+
+  // set COMP rewards
+  amounts[0] = BigInt.fromString(
+    trySupplySpeed.value
+      .toBigDecimal()
+      .times(morphoShare)
+      .times(BLOCKS_PER_DAY.toBigDecimal())
+      .truncate(0)
+      .toString()
+  );
+  amounts[1] = BigInt.fromString(
+    tryBorrowSpeed.value
+      .toBigDecimal()
+      .times(morphoShare)
+      .times(BLOCKS_PER_DAY.toBigDecimal())
+      .truncate(0)
+      .toString()
+  );
```

Comment:
according to the schema documentation, the `rewardTokenEmissionsAmount` is defined as
"Per-block reward token emission as of the current block normalized to a day, in token's native amount. This should be ideally calculated as the theoretical rate instead of the realized amount". 

"token's native amount" is the rewrds token or the market token?

Reply:
That should maybe be more clear, but the reward's token native amount.

So in $COMP in this case